commit b189249637dfc2472bd2a941c59bf2eb98543d68 (refs/changes/81/3481/3)
Author: Mike Zeller <mike.zeller@joyent.com>
Date:   2018-02-26T10:53:22-08:00 (1 year, 7 months ago)
    
    TRITON-179 AddNic should check for existence of ipv4_ips properly
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/nics.js b/lib/nics.js
index e740a12..c91763f 100644
--- a/lib/nics.js
+++ b/lib/nics.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -167,7 +167,11 @@ function addNic(req, res, next) {
         // anyways. If a user passed in a specific ip we will use that.
         assert.uuid(networkArg.ipv4_uuid, 'ipv4_uuid');
         networkUuid = networkArg.ipv4_uuid;
-        networkIp = networkArg.ipv4_ips[0];
+
+        if (networkArg.ipv4_ips) {
+            networkIp = networkArg.ipv4_ips[0];
+        }
+
         break;
     default:
         return next(new InvalidArgumentError(
